Error "g++: error: unrecognized command line option '-mfma'" when building latest master on Yocto (Debian 10) linux

### Describe the bug

I tried to compile latest master sours from repository on Yocto (Debian 10)  Linux 
to get support for WebRTC AEC3 (as describe in https://github.com/pjsip/pjproject/pull/2722)

On 'make dep' step Got Error "g++: error: unrecognized command line option '-mfma'" 


### Steps to reproduce

steps to reproduce:

wget https://github.com/pjsip/pjproject/archive/refs/heads/master.zip  
unzip -a master.zip
cd pjproject-master 
cp pjlib/include/pj/config_site_sample.h pjlib/include/pj/config_site.h
./configure  --disable-video --enable-libwebrtc-aec3 
make dep 

### PJSIP version

latest sources from repositary

### Context

Yocto (Debian 10)  Linux 

### Log, call stack, etc

```shell
h_avx2.cc ../../webrtc_aec3/src//modules/audio_processing/agc2/rnn_vad/vector_math_avx2.S; do \
   if test -f $F; then \
     echo "output/libwebrtc-aec3-aarch64-unknown-linux-gnu/" | tr -d '\n' >> .libwebrtc-aec3-aarch64-unknown-linux-gnu.depend; \
     if echo $F | grep -q "\.c[c|pp]"; then \
        dep="g++ -c -M -Wall -DPJ_AUTOCONF=1 -O2 -DPJ_IS_BIG_ENDIAN=0 -DPJ_IS_LITTLE_ENDIAN=1    -I. -I../../webrtc_aec3/src -I../../../pjlib/include                                     -fexceptions  -DWEBRTC_APM_DEBUG_DUMP=0 -DWEBRTC_POSIX=1 -mfma  -DPJ_AUTOCONF=1 -g -O2 -std=c++17    -Wall -DPJ_AUTOCONF=1 -O2 -DPJ_IS_BIG_ENDIAN=0                                     -DPJ_IS_LITTLE_ENDIAN=1    -I. -I../../webrtc_aec3/src -I../../../pjlib/include $F"; \
     else \
        dep="gcc -c -M -Wall -DPJ_AUTOCONF=1 -O2 -DPJ_IS_BIG_ENDIAN=0 -DPJ_IS_LITTLE_ENDIAN=1    -I. -I../../webrtc_aec3/src -I../../../pjlib/include                                     -fexceptions  -DWEBRTC_APM_DEBUG_DUMP=0 -DWEBRTC_POSIX=1 -mfma $F"; \
     fi; \
     if eval $dep | sed '/^#/d' >> .libwebrtc-aec3-aarch64-unknown-linux-gnu.depend; then \
        true; \
     else \
        echo 'err:' >> .libwebrtc-aec3-aarch64-unknown-linux-gnu.depend; \
        rm -f .libwebrtc-aec3-aarch64-unknown-linux-gnu.depend; \
        exit 1; \
     fi; \
   fi; \
done;
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
gcc: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
gcc: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
g++: error: unrecognized command line option '-mfma'
make[3]: Leaving directory '/home/compulab/pjsip_build_2/pjproject-master/third_party/build/webrtc_aec3'
make[2]: Leaving directory '/home/compulab/pjsip_build_2/pjproject-master/third_party/build/webrtc_aec3'
make[1]: Leaving directory '/home/compulab/pjsip_build_2/pjproject-master/third_party/build'
make[1]: Entering directory '/home/compulab/pjsip_build_2/pjproject-master/pjmedia/build'
make -f /home/compulab/pjsip_build_2/pjproject-master/build/rules.mak APP=PJMEDIA app=pjmedia depend
make[2]: Entering directory '/home/compulab/pjsip_build_2/pjproject-master/pjmedia/build'
rm -f .pjmedia-aarch64-unknown-linux-gnu.depend
for F in  ../src/pjmedia/echo_speex.m ../src/pjmedia/echo_speex.c ../src/pjmedia/echo_speex.cpp ../src/pjmedia/echo_speex.cc ../src/pjmedia/echo_spee                                    x.S  ../src/pjmedia/alaw_ulaw.m ../src/pjmedia/alaw_ulaw.c ../src/pjmedia/alaw_ulaw.cpp ../src/pjmedia/alaw_ulaw.cc ../src/pjmedia/alaw_ulaw.S  ../sr                                    c/pjmedia/alaw_ulaw_table.m ../src/pjmedia/alaw_ulaw_table.c ../src/pjmedia/alaw_ulaw_table.cpp ../src/pjmedia/alaw_ulaw_table.cc ../src/pjmedia/alaw                                    _ulaw_table.S  ../src/pjmedia/avi_player.m ../src/pjmedia/avi_player.c ../src/pjmedia/avi_player.cpp ../src/pjmedia/avi_player.cc ../src/pjmedia/avi_                                    player.S  ../src/pjmedia/bidirectional.m ../src/pjmedia/bidirectional.c ../src/pjmedia/bidirectional.cpp ../src/pjmedia/bidirectional.cc ../src/pjmed
```

Thank you for support, 
commented the lines that you specified in the file is the file: "third_party/build/os-auto.mak.in"
now make dep worked only with 2 errors:

../../webrtc/src/webrtc//modules/audio_processing/aec/aec_core_sse2.c:15:10: fatal error: emmintrin.h: No such file or directory
 #include <emmintrin.h>
          ^~~~~~~~~~~~~
compilation terminated.
../../webrtc/src/webrtc//modules/audio_processing/aec/aec_rdft_sse2.c:13:10: fatal error: emmintrin.h: No such file or directory
 #include <emmintrin.h>
          ^~~~~~~~~~~~~

does that mean that WebRTC was not found? It will be without echo cancellation because of that?

unzipped clean version, applied the patch,
then called configure:
./configure --disable-video --enable-libwebrtc-aec3
and still got the following errors on "make dep"

../../webrtc/src/webrtc//modules/audio_processing/aec/aec_core_sse2.c:15:10: fatal error: emmintrin.h: No such file or directory
 #include <emmintrin.h>
          ^~~~~~~~~~~~~
compilation terminated.
../../webrtc/src/webrtc//modules/audio_processing/aec/aec_rdft_sse2.c:13:10: fatal error: emmintrin.h: No such file or directory
 #include <emmintrin.h>
          ^~~~~~~~~~~~~

I am compiling with WebRTC for echo cancellation, does that mean that I should download and compile WebRTC first? Need instruction how compile with WebRTC, and whether it needs to download WebRTC separately, or part of it is already included in PJSip project?

Downloaded fresh copy of master,
When applying patch got following error:  (so patch was partially applied)
result of patch application:

//*************************************************
compulab@iot-gate-imx8:~/pjsip_build_3/pjproject-master$ patch < aconfigure_patch2.txt
patching file aconfigure
Hunk #1 succeeded at 9243 (offset 3 lines).
Hunk #2 succeeded at 9357 (offset 3 lines).
Hunk #3 FAILED at 9380.
1 out of 3 hunks FAILED -- saving rejects to file aconfigure.rej
patching file aconfigure.ac
Hunk #1 succeeded at 2206 (offset 3 lines).
Hunk #2 succeeded at 2291 (offset 3 lines).

compulab@iot-gate-imx8:~/pjsip_build_3/pjproject-master$ cat aconfigure.rej
--- aconfigure
+++ aconfigure
@@ -9380,7 +9380,6 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
 $as_echo "Checking if libwebrtc-aec3 is enabled...no" >&6; }
                fi

-
 else
   ac_no_webrtc_aec3=1
               $as_echo "#define PJMEDIA_HAS_LIBWEBRTC_AEC3 0" >>confdefs.h

//*************************************************************

then I run:

>./configure --disable-video --enable-libwebrtc-aec3 
>make

make got error:

g++ -c -Wall -DPJ_AUTOCONF=1 -O2 -DPJ_IS_BIG_ENDIAN=0 -DPJ_IS_LITTLE_ENDIAN=1    -I. -I../../webrtc_aec3/src -I../../../pjlib/include -fexceptions -DWEBRTC_ARCH_ARM64 -DWEBRTC_APM_DEBUG_DUMP=0 -DWEBRTC_POSIX=1 -DWEBRTC_HAS_NEON  -DPJ_AUTOCONF=1 -g -O2 -std=c++17    -Wall -DPJ_AUTOCONF=1 -O2 -DPJ_IS_BIG_ENDIAN=0 -DPJ_IS_LITTLE_ENDIAN=1    -I. -I../../webrtc_aec3/src -I../../../pjlib/include \
        -o output/libwebrtc-aec3-aarch64-unknown-linux-gnu/rtc_base/platform_thread_types.o \
        ../../webrtc_aec3/src//rtc_base/platform_thread_types.cc
../../webrtc_aec3/src//rtc_base/platform_thread_types.cc: In function 'rtc::PlatformThreadId rtc::CurrentThreadId()':
../../webrtc_aec3/src//rtc_base/platform_thread_types.cc:46:59: error: invalid cast from type 'pthread_t' {aka 'long unsigned int'} to type 'rtc::PlatformThreadId' {aka 'int'}
   return reinterpret_cast<PlatformThreadId>(pthread_self());
                                                           ^
make[3]: *** [/home/compulab/pjsip_build_3/pjproject-master/build/rules.mak:183: output/libwebrtc-aec3-aarch64-unknown-linux-gnu/rtc_base/platform_thread_types.o] Error 1
make[3]: Leaving directory '/home/compulab/pjsip_build_3/pjproject-master/third_party/build/webrtc_aec3'


make had also some warnings:

../../webrtc_aec3/src//modules/audio_processing/agc2/adaptive_agc.cc: In function 'std::unique_ptr<webrtc::NoiseLevelEstimator> webrtc::{anonymous}::CreateNoiseLevelEstimator(webrtc::{anonymous}::NoiseEstimatorType, webrtc::ApmDataDumper*)':
../../webrtc_aec3/src//modules/audio_processing/agc2/adaptive_agc.cc:57:1: warning: control reaches end of non-void function [-Wreturn-type]

In file included from ../../webrtc_aec3/src/modules/audio_processing/agc2/rnn_vad/rnn_fc.h:19,
                 from ../../webrtc_aec3/src//modules/audio_processing/agc2/rnn_vad/rnn_fc.cc:14:
../../webrtc_aec3/src/api/function_view.h: In instantiation of 'rtc::FunctionView<RetT(ArgT ...)>::FunctionView(F&&) [with F = float (&)(float); typename std::enable_if<std::is_function<typename std::remove_pointer<typename std::remove_reference<_SrcTuple>::type>::type>::value>::type* <anonymous> = 0; RetT = float; ArgT = {float}]':
../../webrtc_aec3/src//modules/audio_processing/agc2/rnn_vad/rnn_fc.cc:58:25:   required from here
../../webrtc_aec3/src/api/function_view.h:79:17: warning: the compiler can assume that the address of 'f' will never be NULL [-Waddress]
       : call_(f ? CallFunPtr<typename std::remove_pointer<F>::type> : nullptr) {
               ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../webrtc_aec3/src/api/function_view.h:79:17: warning: the compiler can assume that the address of 'f' will never be NULL [-Waddress]
../../webrtc_aec3/src//modules/audio_processing/agc2/rnn_vad/rnn_fc.cc: In function 'rtc::FunctionView<float(float)> webrtc::rnn_vad::{anonymous}::GetActivationFunction(webrtc::rnn_vad::ActivationFunction)':
../../webrtc_aec3/src//modules/audio_processing/agc2/rnn_vad/rnn_fc.cc:64:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
In file included from ../../webrtc_aec3/src/modules/audio_processing/agc2/rnn_vad/rnn_fc.h:19,
                 from ../../webrtc_aec3/src//modules/audio_processing/agc2/rnn_vad/rnn_fc.cc:14:
../../webrtc_aec3/src/api/function_view.h: In constructor 'rtc::FunctionView<RetT(ArgT ...)>::FunctionView(F&&) [with F = float (&)(float); typename std::enable_if<std::is_function<typename std::remove_pointer<typename std::remove_reference<_SrcTuple>::type>::type>::value>::type* <anonymous> = 0; RetT = float; ArgT = {float}]':
../../webrtc_aec3/src/api/function_view.h:79:17: warning: nonnull argument 'f' compared to NULL [-Wnonnull-compare]
       : call_(f ? CallFunPtr<typename std::remove_pointer<F>::type> : nullptr) {
               ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

 }

--- a/aconfigure
+++ b/aconfigure
@@ -9246,7 +9246,7 @@ $as_echo "Checking if libwebrtc is disabled...no" >&6; }
                                  ac_webrtc_instset=neon
                                  ac_webrtc_cflags="-DWEBRTC_ARCH_ARMV7 -mfloat-abi=hard -mfpu=neon"
                                  ;;
-			     arm-apple-darwin*)
+			     arm-apple-darwin* | aarch64*)
 				 ac_webrtc_instset=neon
 				 ac_webrtc_cflags="-DWEBRTC_ARCH_ARM64"
 			    	 ;;
@@ -9360,7 +9360,7 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
                                  ac_webrtc_aec3_instset=neon
                                  ac_webrtc_aec3_cflags="-DWEBRTC_ARCH_ARMV7 -mfloat-abi=hard -mfpu=neon"
                                  ;;
-			     arm-apple-darwin*)
+			     arm-apple-darwin* | aarch64*)
 				 ac_webrtc_aec3_instset=neon
 				 ac_webrtc_aec3_cflags="-DWEBRTC_ARCH_ARM64"
 			    	 ;;

--- a/aconfigure.ac
+++ b/aconfigure.ac
@@ -2209,7 +2209,7 @@ AC_ARG_ENABLE(libwebrtc,
                                  ac_webrtc_instset=neon
                                  ac_webrtc_cflags="-DWEBRTC_ARCH_ARMV7 -mfloat-abi=hard -mfpu=neon"
                                  ;;
-			     arm-apple-darwin*)
+			     arm-apple-darwin* | aarch64*)
 				 ac_webrtc_instset=neon
 				 ac_webrtc_cflags="-DWEBRTC_ARCH_ARM64"
 			    	 ;;
@@ -2294,7 +2294,7 @@ AC_ARG_ENABLE(libwebrtc_aec3,
                                  ac_webrtc_aec3_instset=neon
                                  ac_webrtc_aec3_cflags="-DWEBRTC_ARCH_ARMV7 -mfloat-abi=hard -mfpu=neon"
                                  ;;
-			     arm-apple-darwin*)
+			     arm-apple-darwin* | aarch64*)
 				 ac_webrtc_aec3_instset=neon
 				 ac_webrtc_aec3_cflags="-DWEBRTC_ARCH_ARM64"
 			    	 ;;

